waspc/data/Generator/templates/server/src/dbClient.js

```diff
@@ -12,6 +13,41 @@ const createDbClient = () => {
   const prismaClient = new Prisma.PrismaClient()
 
   {=# isAuthEnabled =}
+
+  prismaClient.$use(async (params, next) => {
+    // Ensure strong plaintext password. Must come before hashing middleware.
+    // Throws an EntityValidationError on the first validation that fails.
+    if (params.model === '{= userEntityUpper =}' && params.action === 'create') {
+        const data = params.args.data || {}
+
+        const defaultValidations = [
+          { name: 'email must be present', fn: data => !!data.email },
+          { name: 'password must be present', fn: data => !!data.password },
+          { name: 'password must be at least 8 characters', fn: data => data.password.length >= 8 },
+          { name: 'password must contain a number', fn: data => /\d/.test(data.password) },
+        ]
+
+        let validations = params.args._waspSkipDefaultValidations ? [] : defaultValidations
+        if (Array.isArray(params.args._waspCustomValidations)) {
+          validations = validations.concat(params.args._waspCustomValidations)
+        }
+
+        for (const validation of validations) {
+          if (!validation.fn(data)) {
+            throw new EntityValidationError('{= userEntityUpper =}', validation.name)
+          }
+        }
+
+        // Remove from downstream Prisma processing to avoid "Unknown arg" error
+        delete params.args._waspSkipDefaultValidations
+        delete params.args._waspCustomValidations
+    }
+
+    const result = next(params)
+
+    return result
+  })
+
```

Comment:
Looking at this whole file, now we have two middlewares that are focusing on authentication, on user enityt. Should we consider extracting that code, for example into something like `src/core/auth/...` (not sure about exact location, needs some more consideration) and then pulling it in from there and just registering it as middleware here? In the future we can expect to have some more middleware here, so that way we keep it clean and organized by the domain (in this case auth).

Reply:
Yeah that is a good thought, thanks. With just the original 1 it was not such a big deal, but with 2 it is making it messier. May be nice to have a spot for auth middleware and then just attach it to the dbclient (and form a pattern for similar concerns).

Reply:
Yes, exactly! This might be a good moment to do this? We can do something simple, and we can try to be smarter in the future about the pattern itself, but at least we will get a good start.

Reply:
This has been addressed here: https://github.com/wasp-lang/wasp/pull/376/commits/3ffbc62c75cf4a43363b90fc8c2b1480309891ba Thanks!